gowrapper/main.go

```diff
@@ -215,24 +226,95 @@ func convertViaJSON(in interface{}, out interface{}) error {
 	return json.Unmarshal(jsonBytes, out)
 }
 
+type uint64AsDecimalString uint64
+
+func (u *uint64AsDecimalString) UnmarshalJSON(bytes []byte) error {
+	var str string
+	if err := json.Unmarshal(bytes, &str); err != nil {
+		return err
+	}
+	int, ok := new(big.Int).SetString(str, 10)
+	if !ok {
+		return errors.New("expected decimal string as value")
+	}
+	*u = uint64AsDecimalString(int.Uint64())
+	return nil
+}
+
+type btcSignInputRequest struct {
+	PrevOutHash  []byte                `json:"prevOutHash"`
+	PrevOutIndex uint32                `json:"prevOutIndex"`
+	PrevOutValue uint64AsDecimalString `json:"prevOutValue"`
+	Sequence     uint32                `json:"sequence"`
+	Keypath      []uint32              `json:"keypath"`
+}
+
+func (input *btcSignInputRequest) toInput() *messages.BTCSignInputRequest {
+	return &messages.BTCSignInputRequest{
+		PrevOutHash:  input.PrevOutHash,
+		PrevOutIndex: input.PrevOutIndex,
+		PrevOutValue: uint64(input.PrevOutValue),
+		Sequence:     input.Sequence,
+		Keypath:      input.Keypath,
+	}
+}
+
+type btcSignOutputRequest struct {
+	Ours    bool                   `json:"ours"`
+	Type    messages.BTCOutputType `json:"type"`
+	Value   uint64AsDecimalString  `json:"value"`
+	Hash    []byte                 `json:"hash"`
+	Keypath []uint32               `json:"keypath"`
+}
+
+func (output *btcSignOutputRequest) toOutput() *messages.BTCSignOutputRequest {
+	return &messages.BTCSignOutputRequest{
+		Ours:    output.Ours,
+		Type:    output.Type,
+		Value:   uint64(output.Value),
+		Hash:    output.Hash,
+		Keypath: output.Keypath,
+	}
+}
+
+func convertInputsAndOutputs(
```

Comment:
Same as for `demo.js`, would be good to move some of this out to some util file from `main.go`